### PR TITLE
Add support for the sync endpoint

### PR DESCRIPTION
**Pull request by trodrigues**

### CODE REVIEW

File: test/integration/sync-test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+buster.testCase('=> Sync', {
```
Comment (sdepold) [inline test/integration/sync-test.js]: Please remove the focus rocket

File: index.js [hunk 2]
```diff
@@ -90,6 +90,33 @@ var Client = redefine.Class({
...
+                   return {
+                     items: sync.items,
+                     nextSyncToken: sync.nextSyncToken
+                   };
+                 }
+               });
+  },
+
+  sync: function (object, callback) {
+    var query = Query.parse(object);
```
Comment (sdepold) [inline index.js]: Shouldn't we do any kind of sanity check here?

File: index.js [hunk 3]
```diff
@@ -90,6 +90,33 @@ var Client = redefine.Class({
...
+    var items = [];
```
Comment (sdepold) [inline index.js]: What do we need this variable for ?

File: test/integration/sync-test.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  prepare: function() {
+    this.timeout = 90e3;
+  },
+
+  'can be read': function() {
```
Comment (sdepold) [inline test/integration/sync-test.js]: Could we add a test that does the initial sync, adds an entry and continuous the sync with the nextSyncUrl, which should then return the just created entry?

**Comment by sdepold** [comment]
I added a change log in master. Please rebase and add your notes there :)

File: test/integration/sync-test.js (hunk 1 above)
Comment (trodrigues) [inline test/integration/sync-test.js]: oops!

File: index.js (hunk 2 above)
Comment (trodrigues) [inline index.js]: good point

File: index.js (hunk 3 above)
Comment (trodrigues) [inline index.js]: Ah good catch. I wrote an initial version of the code that was very different, then refactored stuff into the Sync class and this is somehow a leftover.

File: test/integration/sync-test.js (hunk 4 above)
Comment (trodrigues) [inline test/integration/sync-test.js]: Yeah, I thought about a way of testing the next sync but there's no good way of doing that.

**Comment by sdepold** [comment]
LGTM

**Pull request merged**